### DISCUSSION TITLE
Support for Java 17+

### DISCUSSION THREAD

**Discussion by yannickkirschen** [Pi4J/pi4j#295] [General]
Hey there, I can't compile my project using Java 17 but it works just fine with 17. In your examples you are also using Java 11. Do you plan to support Java 17+ in the future or is this just my fault?

**Comment by FDelporte** [comment]
Hi, I have used all versions since 11 with Pi4J without any issues...

**Reply by yannickkirschen** [reply]
That's good to know! When using Java 17, I get `Fatal error compiling: error: release version 17.0.6 not supported` when compiling. This works when I set 11 in the pom. Do you have configured something special?

**Reply by FDelporte** [reply]
Could it be you are on an old Maven version? Or Maven is using older Java?

Check with `mvn -v`

**Reply by yannickkirschen** [reply]
No, that looks all good:

```
Maven home: /opt/homebrew/Cellar/maven/3.9.4/libexec
Java version: 20.0.1, vendor: Homebrew, runtime: /opt/homebrew/Cellar/openjdk/20.0.1/libexec/openjdk.jdk/Contents/Home
Default locale: en_DE, platform encoding: UTF-8
OS name: "mac os x", version: "13.5.1", arch: "aarch64", family: "mac"
```

(btw: I set Java 17 from IntelliJ; when executing `mvn clean package` directly, it uses 20.0.1 and fails with the same error)

**Reply by yannickkirschen** [reply]
It suddenly works! I removed some plugins from the example I didn't need and now it compiles. Thanks for your help!
